Source/Core/Common/Logging/LogManager.h

```diff
@@ -79,7 +81,7 @@ class LogManager
 
   LogLevel m_level;
   EnumMap<LogContainer, LAST_LOG_TYPE> m_log{};
-  std::array<LogListener*, LogListener::NUMBER_OF_LISTENERS> m_listeners{};
+  std::array<std::unique_ptr<LogListener>, LogListener::NUMBER_OF_LISTENERS> m_listeners{};
```

Comment:
I don't know if `unique_ptr` here or `LogListenerImpl` in general is gaining you anything? `LogListenerImpl` just holds a raw pointer to the `LogWidget` so you haven't extended the life or whatever and might as well just hold a raw-pointer like before?

Reply:
I haven't gained anything regarding LogListenerImpl, but I have eliminated the `new` and `delete` for FileLogListener and ConsoleListener.

Reply:
Oh, I see, `LogListenerImpl` is an adapter to make a non-owning pointer in a situation that's normally owning.